discord/gateway.py

```diff
@@ -142,6 +147,20 @@ def get_payload(self):
             'd': int(time.time() * 1000)
         }
 
+    def ack(self):
+        ack_time = time.perf_counter()
+        self._last_ack = ack_time
+        self.recent_ack_latencies.append(ack_time - self._last_send)
```

Comment:
What is `_last_send`?

Reply:
`_last_send` is the `time.perf_counter()` of the last `VoiceKeepAliveHandler` heartbeat send, inherited from parent: `KeepAliveHandler`.